lib/App/DuckPAN/Web.pm

```diff
@@ -285,11 +284,21 @@ sub request {
 					# Inject a script which prints out what we want.
 					# There is no error-checking or support for non-auto-templates here.
 					my $structured = $result->structured_answer;
-					my $template_name = 'goodie_'.scalar @{$structured->{input}}.'_inputs';
-					my $json_string = encode_json({Answer => $structured});
-					$zci_container->push_content(HTML::TreeBuilder->new_from_content("<script>\$(window).load(function(){"
-								. "document.getElementById('zci-answer').innerHTML = DDG.exec_template('$template_name', $json_string);"
-								. "});</script>")->guts);
+					if(exists $structured->{templates}){ # user-specified templates
+						$page = $root->as_HTML;
+						$page =~ s|####DUCKDUCKHACK-CALL-NRJ####|;DDG.duckbar.future_signal_tab({signal:'high',from:'$structured->{id}'});|;
```

Comment:
We actually do this below starting at line 359. If you just create the script tag and it into the correct arrat, e.g. `@calls_nrj` this should work the same and then you don't have to both modifying the page here I believe?